### PR TITLE
Implement Section 461(l) excess business loss limitation

### LINKED ISSUE #7015: Partnership/S-corp losses not included in AGI; missing Section 461(l) excess business loss limitation
## Summary

PolicyEngine does not include partnership/S-corp losses in adjusted gross income. The `irs_gross_income` variable caps all income sources at 0 (line 19: `max_(0, ...)`), and `loss_ald` only handles self-employment and capital losses—not partnership/S-corp losses. Additionally, the Section 461(l) excess business loss limitation is not implemented.

### 1. S-corp losses ignored in gross income

In `policyengine_us/variables/gov/irs/income/taxable_income/adjusted_gross_income/irs_gross_income/irs_gross_income.py`:

```python
def formula(person, period, parameters):
    sources = parameters(period).gov.irs.gross_income.sources
    total = 0
    not_dependent = ~person("is_tax_unit_dependent", period)
    for source in sources:
        # Add positive values only - losses are deducted later.
        total += not_dependent * max_(0, add(person, period, [source]))  # <-- caps at 0
    return total
```

### 2. S-corp losses not deducted in loss_ald

In `policyengine_us/variables/gov/irs/income/taxable_income/adjusted_gross_income/above_the_line_deductions/loss_ald.py`:

```python
def formula(tax_unit, period, parameters):
    # Only handles self_employment_income - NOT partnership_s_corp_income
    indiv_se_loss = max_(0, -person("self_employment_income", period))
    self_employment_loss = tax_unit.sum(indiv_se_loss)
    limited_capital_loss = tax_unit("limited_capital_loss", period)
    return min_(max_loss, self_employment_loss + limited_capital_loss)
```

### 3. Section 461(l) not implemented

IRC §461(l) limits excess business losses for noncorporate taxpayers. For 2024:
- **Single/Other**: $305,000
- **MFJ**: $610,000

Losses exceeding this threshold become NOL carryforwards.

## Legal Reference

[26 U.S.C. §461(l)](https://www.law.cornell.edu/uscode/text/26/461):

> **(1) Limitation** — In the case of a taxpayer other than a corporation, any excess business loss of the taxpayer for the taxable year shall not be allowed.
>
> **(3) Excess business loss** — The term "excess business loss" means the excess (if any) of—
> - (i) the aggregate deductions attributable to trades or businesses... over
> - (ii) the sum of aggregate gross income/gain from trades or businesses, plus $250,000 ($500,000 MFJ, indexed for inflation)

## Example (from policyengine-taxsim issue #674)

**Input:**
- MFJ, ages 26 and 27
- Spouse wages: $82,813
- S-corp loss: -$838,974
- Capital losses: -$224,839
- Interest: $20

**PE calculates:**
- Federal AGI: $79,833 (S-corp loss completely ignored)
- Only $3,000 capital loss deducted

**TaxAct (correct) calculates:**
- S-corp loss on Schedule E: -$838,974
- Form 461 limitation: $610,000 threshold
- Excess business loss add-back: +$228,974
- Federal AGI: **-$530,137**

## Files to Fix

1. `policyengine_us/variables/gov/irs/income/taxable_income/adjusted_gross_income/above_the_line_deductions/loss_ald.py`
   - Add `partnership_s_corp_income` losses to the deduction

2. New file needed for Section 461(l):
   - `policyengine_us/variables/gov/irs/income/taxable_income/adjusted_gross_income/excess_business_loss_limitation.py`
   - Implement Form 461 calculation

3. New parameters needed:
   - `policyengine_us/parameters/gov/irs/ald/loss/excess_business_loss_threshold.yaml`
   - 2024: $305,000 (single) / $610,000 (MFJ)

### Step 1: Add S-corp losses to loss_ald.py

```python
def formula(tax_unit, period, parameters):
    filing_status = tax_unit("filing_status", period)
    max_loss = parameters(period).gov.irs.ald.loss.max[filing_status]
    person = tax_unit.members
    
    # Self-employment losses
    indiv_se_loss = max_(0, -person("self_employment_income", period))
    self_employment_loss = tax_unit.sum(indiv_se_loss)
    
    # Partnership/S-corp losses (NEW)
    indiv_scorp_loss = max_(0, -person("partnership_s_corp_income", period))
    partnership_s_corp_loss = tax_unit.sum(indiv_scorp_loss)
    
    # Capital losses
    limited_capital_loss = tax_unit("limited_capital_loss", period)
    
    # Apply Section 461(l) limitation to business losses
    total_business_loss = self_employment_loss + partnership_s_corp_loss
    excess_business_loss_threshold = parameters(period).gov.irs.ald.loss.excess_business_loss_threshold[filing_status]
    limited_business_loss = min_(total_business_loss, excess_business_loss_threshold)
    
    return min_(max_loss, limited_business_loss + limited_capital_loss)
```

### Step 2: Add threshold parameter

```yaml
# excess_business_loss_threshold.yaml
description: Section 461(l) excess business loss threshold for noncorporate taxpayers.
metadata:
  reference:
    - title: 26 U.S.C. §461(l)(3)(A)(ii)(II)
      href: https://www.law.cornell.edu/uscode/text/26/461
    - title: Form 461 Instructions
      href: https://www.irs.gov/pub/irs-pdf/i461.pdf
  unit: currency-USD
  period: year
  label: Excess business loss threshold
SINGLE:
  2021-01-01: 262000
  2022-01-01: 270000
  2023-01-01: 289000
  2024-01-01: 305000
JOINT:
  2021-01-01: 524000
  2022-01-01: 540000
  2023-01-01: 578000
  2024-01-01: 610000
# Other filing statuses use single amount
```

## Integration Test

```yaml
- name: S-corp loss with Section 461(l) limitation
  period: 2024
  absolute_error_margin: 1
  input:
    people:
      person1:
        age: 26
        partnership_s_corp_income: -419487
        short_term_capital_gains: -111233
        long_term_capital_gains: -1187
        taxable_interest_income: 10
        is_tax_unit_head: true
      person2:
        age: 27
        employment_income: 82813
        partnership_s_corp_income: -419487
        short_term_capital_gains: -111233
        long_term_capital_gains: -1187
        taxable_interest_income: 10
        is_tax_unit_spouse: true
    tax_units:
      tax_unit:
        members: [person1, person2]
    spm_units:
      spm_unit:
        members: [person1, person2]
    households:
      household:
        members: [person1, person2]
        state_fips: 44
  output:
    adjusted_gross_income: -530137  # Negative AGI after 461(l) limitation
```

## Related Issues

- PolicyEngine/policyengine-taxsim#674: RI joint 82Kpwages -838K psemp

### PR DESCRIPTION
**Pull request by MaxGhenis**
## Summary
- S-corp/partnership losses now included in AGI calculation via `loss_ald`
- Section 461(l) excess business loss limitation correctly applied to business losses (self-employment + partnership/S-corp)
- Capital losses handled separately under Section 1211 ($3,000 limit)

## Changes
- **Modified** `policyengine_us/variables/gov/irs/income/taxable_income/adjusted_gross_income/above_the_line_deductions/loss_ald.py`:
  - Added partnership/S-corp losses to business loss calculation
  - Applied Section 461(l) threshold to business losses only (not capital losses)
  - Capital losses continue to be limited separately by Section 1211

- **Added** `policyengine_us/tests/policy/baseline/gov/irs/income/taxable_income/adjusted_gross_income/above_the_line_deductions/loss_ald.yaml`:
  - Integration test from issue (MFJ with $838K S-corp loss)
  - Self-employment loss under threshold (single filer)
  - Combined SE and S-corp loss exceeding threshold
  - Business losses plus capital losses (separate limits)
  - MFJ higher threshold test
  - Section 461(l) sunset after 2026 test

## Test
- [x] All 6 new loss_ald tests pass
- [x] All 189 IRS income tests pass
- [x] All 16 IRS integration tests pass (including capital_gains_losses.yaml)

Closes #7015

### CODE REVIEW

**Comment by MaxGhenis** [comment]
Closing — this PR has had merge conflicts for 2+ weeks. Happy to revive if the work is still needed; reopen after rebasing from `upstream/main`.
